src/cascade/model/model.py

```diff
@@ -0,0 +1,115 @@
+from cascade.dismod.constants import WeightEnum
+from cascade.model.dismod_groups import DismodGroups
+from cascade.model.var import Var
+from cascade.model.smooth_grid import uninformative_grid_from_var
+
+
+class Model(DismodGroups):
+    """
+    A DismodGroups container of SmoothGrid.
+
+    Uses locations as given and translates them into nodes for Dismod-AT.
+    Uses ages and times as given and translates them into ``age_id``
+    and ``time_id`` for Dismod-AT.
+    """
+    def __init__(self, nonzero_rates, parent_location, child_location, covariates=None, weights=None):
+        """
+        >>> locations = location_hierarchy(execution_context)
+        >>> m = Model(["chi", "omega", "iota"], 6, locations)
+        """
+        super().__init__()
+        self.nonzero_rates = nonzero_rates
+        self.location_id = parent_location
+        self.child_location = child_location
+        # Covariates are here because their reference values are part of
+        # the model. Even though avgint and data use them, a model is always
+        # written before the avgint and data are written.
+        self.covariates = covariates if covariates else list()
+        # There are always four weights, constant, susceptible,
+        # with_condition, and total.
+        if weights:
+            self.weights = weights
+        else:
+            self.weights = dict()
```

Comment:
If there are always four, should this be constrained to a type that knows that:
```
class Weights:
    __slots__ = ["constant", "susceptible", "with_condition"]
...
```
Then you could push some of the checks in `_ensure_weights` down into that

Reply:
Lots of checks are missing in this PR. I was very intentionally light on checks at the API interface until we agree on the interface. And I'm the one making up this rule about the four weights. The modelers should sign off on it.